### PR TITLE
Implements Session.destroy method which stops sending bosh requests and aborts the current requests

### PR DESCRIPTION
**Pull request by hristoterezov**
Moved from https://github.com/node-xmpp/client/pull/203.

### CODE REVIEW

**Comment by sonnyp** [comment]
@hristoterezov thanks!

File: packages/node-xmpp-client/lib/bosh.js [hunk 1]
```diff
@@ -180,15 +194,18 @@ BOSHConnection.prototype.request = function (attrs, children, cb, retry) {
...
     headers: { 'Content-Type': this.contentType },
     body: boshEl.toString()
   },
     function (err, res, body) {
       that.currentRequests--
-
+      var currentRequestIndex = that.activeRequests.indexOf(currentRequest)
+      if (currentRequestIndex != -1) {
+        that.activeRequests.splice(currentRequestIndex, 1)
+      }
```
Comment (sonnyp) [inline packages/node-xmpp-client/lib/bosh.js]: maybe a Set rather than an Array for currentRequests, what do you think?

Comment (hristoterezov) [inline packages/node-xmpp-client/lib/bosh.js]: Fixed!

File: packages/node-xmpp-client/lib/bosh.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 'use strict'
 
+require('es6-collections')
```
Comment (sonnyp) [inline packages/node-xmpp-client/lib/bosh.js]: no need, we don't support Node.js 0.10 anymore

File: packages/node-xmpp-client/package.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
     "minimist": "^1.2.0",
     "node-xmpp-core": "^5.0.0",
     "request": "^2.65.0",
-    "ws": "^1.1.0"
+    "ws": "^1.1.0",
+    "es6-collections": "^0.5.6"
```
Comment (sonnyp) [inline packages/node-xmpp-client/package.json]: ditto

File: packages/node-xmpp-client/lib/bosh.js (hunk 2 above)
Comment (hristoterezov) [inline packages/node-xmpp-client/lib/bosh.js]: Fixed! 

I noticed that you have this require in node-xmpp-server. Probably you want to remove it .

File: packages/node-xmpp-client/lib/bosh.js [hunk 4]
```diff
@@ -180,15 +194,15 @@ BOSHConnection.prototype.request = function (attrs, children, cb, retry) {
...
+  var currentRequest = request({
     uri: this.boshURL,
     method: 'POST',
     headers: { 'Content-Type': this.contentType },
     body: boshEl.toString()
   },
     function (err, res, body) {
       that.currentRequests--
-
+      this.activeRequests.delete(currentRequest);
```
Comment (sonnyp) [inline packages/node-xmpp-client/lib/bosh.js]: that, not this :)

**Comment by sonnyp** [comment]
https://github.com/node-xmpp/node-xmpp/pull/322

**Pull request closed without merging**